src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -678,7 +678,14 @@ trait Namers extends MethodSynthesis {
             // which could upset other code paths)
             if (!scopePartiallyCompleted)
               companionContext.scope.unlink(sym)
+
+            for (a <- sym.attachments.get[CaseApplyDefaultGetters]; defaultGetter <- a.defaultGetters) {
+              companionContext.unit.synthetics -= defaultGetter
+              companionContext.scope.unlink(defaultGetter)
+            }
           }
+
+          sym.removeAttachment[CaseApplyDefaultGetters]
```

Comment:
It's not clear to me why we remove the attachment? I guess it's no longer needed, but we could start using it for other purposes later, so it would be good to add a comment to make it clear it doesn't *have* to be removed for correctness.